presto-main/src/main/java/com/facebook/presto/execution/QueryState.java

```diff
@@ -23,42 +23,52 @@
     /**
      * Query has been accepted and is awaiting execution.
      */
-    QUEUED(false),
+    QUEUED(false, true),
     /**
      * Query is being planned.
      */
-    PLANNING(false),
+    PLANNING(false, true),
```

Comment:
The `PLANNING` and `STARTING` states means the query is actually running (from the user's perspective). These states may take substantial time, depending on the working being done, but the work is actually happening.